File: Module.Frontend.TPM/Client/model/tpm/postpromoeffect/PostPromoEffect.js
Ext.define('App.model.tpm.postpromoeffect.PostPromoEffect', {
    extend: 'Ext.data.Model',
    idProperty: 'Id',
    breezeEntityType: 'PostPromoEffect',
    fields: [
        { name: 'Id', hidden: true },
        { name: 'StartDate', type: 'date', hidden: false, isDefault: true },
        { name: 'EndDate', type: 'date', hidden: false, isDefault: true },
        { name: 'ClientTreeId', hidden: true, isDefault: false },
        { name: 'ProductTreeId', hidden: true, isDefault: false },
        {
            name: 'ClientTreeFullPathName', type: 'string', mapping: 'ClientTree.FullPathName', tree: true,
            defaultFilterConfig: { valueField: 'FullPathName' }, breezeEntityType: 'ClientTree', hidden: false, isDefault: true
        },
        {
            name: 'ProductTreeFullPathName', type: 'string', mapping: 'ProductTree.FullPathName', tree: true,
            defaultFilterConfig: { valueField: 'FullPathName' }, breezeEntityType: 'ProductTree', hidden: false, isDefault: true
        },
        { name: 'EffectWeek1', type: 'float', hidden: false, isDefault: true, defaultValue: 0 },
        { name: 'EffectWeek2', type: 'float', hidden: false, isDefault: true, defaultValue: 0 },
        { name: 'TotalEffect', type: 'float', hidden: false, isDefault: true, defaultValue: 0 },
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'PostPromoEffects',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    }
});

File: Module.Persist.TPM/Model/TPM/PostPromoEffect.cs
using Core.Data;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Module.Persist.TPM.Model.TPM
{
    public class PostPromoEffect : IEntity<Guid>, IDeactivatable
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public System.Guid Id { get; set; }
        public bool Disabled { get; set; }
        public DateTimeOffset? DeletedDate { get; set; }

        public DateTimeOffset StartDate { get; set; }
        public DateTimeOffset EndDate { get; set; }
        public int ClientTreeId { get; set; }
        public int ProductTreeId { get; set; }
        public double EffectWeek1 { get; set; }
        public double EffectWeek2 { get; set; }
        public double TotalEffect { get; set; }

        public virtual ClientTree ClientTree { get; set; }
        public virtual ProductTree ProductTree { get; set; }
    }
}
